src/codeu/chat/server/Server.java

```diff
@@ -206,7 +246,125 @@ public void run() {
         timeline.scheduleIn(RELAY_REFRESH_MS, this);
       }
     });
-    
+  }
+
+  // This method will write the contents of dataList to a txt file
+  public void saveServer() {
```

Comment:
Is `saveServer()` thread-safe? You're reading the same objects which can be modified by another thread at the same time.